python/ray/serve/_private/http_proxy.py

```diff
@@ -1374,34 +1388,83 @@ def __init__(
                 request_timeout_s or RAY_SERVE_REQUEST_PROCESSING_TIMEOUT_S
             ),
         )
+        self.grpc_proxy = (
```

Comment:
Right now there is no way to turn off HTTP proxy by itself. Although we have the `proxy_location` config, that's more gear towards the proxy in general instead of focusing on HTTP proxy. I agreed most users probably won't be using both HTTP and gRPC in the same service, but this is nontrivial to start serve with only gRPC proxy and not HTTP proxy. For one, this will for sure cause issue running on product without the product change already deployed. There can be a huge impact if we decide not to start HTTP proxy. We can do a followup to better research on starting only gRPC proxy, but for now I don't think there is quick and easy solution for it.

Reply:
Why is that non-trivial? If i'm omitting any HTTP configuration -- that's a clear signal i don't need an HTTP path.

No need to do it in this PR, we can do it in a follow-up.